Ethans branch

hille7 opened the pull request:
Hi Dr. Butler,

Here is my pull request for my Iris-repo branch that contains my homework submissions!

Thanks,
Ethan

hille7 committed:
add script to make 4 scatter plots

hille7 committed:
Add script to make bivariate plots

hille7 committed:
Add script to make bivariate plots

hille7 committed:
Add script to make bivariate plots

hille7 committed:
this is better than claire's

hille7 committed:
this is better than claire's

hille7 committed:
this is better than claire's

hille7 committed:
Merge branch 'master' of https://github.com/Rbootcamp-UHM/Iris-repo into ethans-branch

hille7 committed:
Here are some new .Rmd files

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Homework 1

hille7 committed:
Delete ethans_iris_plots.R

hille7 committed:
Homework 1

hille7 committed:
Merge branch 'ethans-branch' of https://github.com/Rbootcamp-UHM/Iris-repo into ethans-branch

hille7 committed:
Delete markdown_ethan_test.Rmd

hille7 committed:
Delete Markdown_ethan_test.html

hille7 committed:
Homework 1

hille7 committed:
Merge branch 'ethans-branch' of https://github.com/Rbootcamp-UHM/Iris-repo into ethans-branch

hille7 committed:
Homework 1

mbutler808 commented on iris_data_explore.R:
```diff
@@ -0,0 +1,71 @@
+
+git add iris_data_explore.R
+git commit -m "Add script to make bivariate plots"
+git push origin ethans-branch
```
All of this causes an error! Please comment out
Otherwise, very nice script :)

mbutler808 replied:
Unfortunately, these lines were not commented out. All files must run without error -1.

mbutler808 commented on iris_markdown_EH.Rmd:
````diff
@@ -0,0 +1,123 @@
+
+```{r}
+setwd("C:/Users/hille/Documents/Rbootcamp_S2020/Iris-repo/")
+```
````
True, but donʻt forget to comment it out in any script. You can also save an empty x.Rdata file (in windows) or x.R (in mac) in the appropriate directory and start from there.

mbutler808 replied:
Unfortunately, this also didnʻt get fixed. Normally, in a coding class you would get a zero for a program that doesnʻt even run.  -1

mbutler808 commented on iris_markdown_EH.Rmd:
```diff
@@ -0,0 +1,123 @@
+
+
+Next, I've scrambled the iris dataset (gotta make it harder) and stored the scrambled data into the object called mix:
+
```
lol

mbutler808 commented on iris_markdown_EH.Rmd:
````diff
@@ -0,0 +1,123 @@
+```
+
+
+Once I have the scrambled data stored in the object called "mix" I can store each of the columns as a seperate vector:
````
This is really repetitive. You state what youʻre going to do, then after you did it you repeat it and say Next.... Can you please cut repetition?

mbutler808 commented on iris_markdown_EH.Rmd:
```diff
@@ -0,0 +1,123 @@
+color2[species2 == "versicolor"] <- "blue"
+color2
+color2[species2 == "virginica"] <- "orange"
+color2
```
Not wrong, but what is the purpose of species2? Does its value ever change? If not, you might want to cut this object.

mbutler808 commented on iris_markdown_EH.Rmd:
````diff
@@ -0,0 +1,123 @@
+```
+
+Cool, Now we can start doing some real analyses. 
+I have have made a best fit line of regression for Sepel Width as a function of Petal Length and have displayed the output stats:
````
sepal - please fix spelling throughout

mbutler808 commented on iris_markdown_EH.Rmd:
````diff
@@ -0,0 +1,123 @@
+legend('top', legend = levels(species), col = c("purple", "blue", "orange"), cex = .8, pch = 19)
+```
+
+As we can see from the graph above, the "general" trend is that as petal length increases, the sepal width decreases.
+
+BUT WAIT THERE'S MORE!!!!
````
Very nice!!! And I love this 👍

mbutler808 commented on iris_markdown_EH.Rmd:
```diff
@@ -0,0 +1,123 @@
+yhat.setosa <- slope.setosa*setosa$Petal.Length+int.setosa
+yhat.setosa
+error.setosa.sepalW <- setosa$Sepal.Width-yhat.setosa
+error.setosa.sepalW
```
This is very nice!!!  I love your idea to compute the regression separately by species.
But FYI if you just want to plot the line of best fit, there is already a function for that:
abline( setosaPLvsSW.lm)

mbutler808 commented on iris_markdown_EH.Rmd:
```diff
@@ -0,0 +1,123 @@
+
+What we can see from THIS plot is that, within species, as petal length increases, sepal width ALSO increases. WOW, who would have thought.
+
+And this will conclude my simple anaylsis of the iris dataset!
+
```
EXCELLENT JOB!!!! 💯

mbutler808 replied:
The markdown doesnʻt run by itself, even without the working directory commented out.  It does work when it is run through the script, "iris_simpleanalysis_EH.R", but it produces an error when it is run alone:

> require(rmarkdown)
Loading required package: rmarkdown
> require("rmarkdown")
> #Sys.setenv(RSTUDIO_PANDOC="C:/Program Files/RStudio/bin/pandoc")
> render("iris_markdown_EH.Rmd", "html_document")


processing file: iris_markdown_EH.Rmd
  |....                                                           |   7%
  |........                                                       |  13%
  |.............                                                  |  20%
  |.................                                              |  27%
  |.....................                                          |  33%
  |.........................                                      |  40%
  |.............................                                  |  47%
  |..................................                             |  53%
  |......................................                         |  60%
  |..........................................                     |  67%
  |..............................................                 |  73%
  |..................................................             |  80%
  |.......................................................        |  87%
  |...........................................................    |  93%
label: unnamed-chunk-7
Quitting from lines 113-118 (iris_markdown_EH.Rmd) 
Error in abline(setosaPLvsSW.lm, col = "purple") : 
  object 'setosaPLvsSW.lm' not found

Looks like you forgot to copy a line or two of code into your markdown, but itʻs in your script.  Please make sure you do proper error testing in future.  -M

mbutler808 reviewed (changes requested):
VERY nice job ETHAN!!!! I love your ideas. Just some minor comments. Can you please take a look and correct, if necessary?
Marguerite

hille7 committed:
Homework 1 Revised

mbutler808 reviewed (commented):
Iʻm sad that you didnʻt make the changes. You need to be a bit more diligent. This code wonʻt work as written.  See comments below. 

Marguerite

hille7 committed:
Delete iris_data_explore.R

hille7 committed:
Homework 1 Revised

hille7 committed:
Merge branch 'ethans-branch' of https://github.com/Rbootcamp-UHM/Iris-repo into ethans-branch

hille7 commented:
Hey Dr. Butler,

Thank you again for all the clarification regarding how Rmarkdown works and how it should be independent of the "R" script. It was extremely helpful. 

I think I corrected all of the errors in the markdown file and now it should be able to run independently of the Rscript. I removed the "setwd()" command in the beginning and I made the explanations less repetitive. I have also refined object names ("species2" served no purpose so I changed that to just "species" etc.). I have also found why the "Rmd" file threw an error; I had a "]" instead of a "}" in one of the "```{r}" lines (tiny typos can make all the difference lol). I hope it works without a hitch this time (my fingers are very crossed).

-Ethan
